widget/widget.go

```diff
@@ -163,6 +169,15 @@ func (w *BaseWidget) super() fyne.Widget {
 	return impl
 }
 
+// theme returns a cached theme instance for this widget (or its extending widget).
+func (w *BaseWidget) theme() fyne.Theme {
```

Comment:
I wonder if this would be better off exported for other developers to enjoy?

Reply:
I did wonder about that. But if we do, people might thing that overriding it will use a different theme - so I wanted to avoid that confusion.

Reply:
Hmm. I don't know. Seems like a very useful thing to have access to.

Reply:
That's fair - I added it as a public API and tried to be clear that it's a utility.